tough/src/editor/signed.rs

```diff
@@ -262,22 +297,20 @@ impl SignedRepository {
         // Use the file name to see if a target exists in the repo
         // with that name. If so...
         let repo_targets = &self.targets.signed.signed.targets;
-        if let Some(repo_target) = repo_targets.get(file_name) {
-            // compare the hashes of the target from the repo and the
-            // target we just created. If they are the same, this must
-            // be the same file, symlink it.
-            ensure!(
-                target_from_path.hashes.sha256 == repo_target.hashes.sha256,
-                error::HashMismatch {
-                    context: "target",
-                    calculated: hex::encode(target_from_path.hashes.sha256),
-                    expected: hex::encode(&repo_target.hashes.sha256),
-                }
-            );
-        } else {
-            // Given path is not a target; caller can decide if that's a problem
-            return Ok(None);
-        }
+        let repo_target = repo_targets
+            .get(file_name)
+            .context(error::PathIsNotTarget { path: input })?;
+        // compare the hashes of the target from the repo and the
+        // target we just created. If they are the same, this must
+        // be the same file, symlink it.
+        ensure!(
+            target_from_path.hashes.sha256 == repo_target.hashes.sha256,
+            error::HashMismatch {
+                context: "target",
+                calculated: hex::encode(target_from_path.hashes.sha256),
+                expected: hex::encode(&repo_target.hashes.sha256),
+            }
+        );
```

Comment:
nit: it's a bit odd to find "symlink it" in a comment here when the symlink happens much later